include/pagmo/algorithms/mbh.hpp

```diff
@@ -0,0 +1,374 @@
+     * tolerance) and Trial is the trial number (which will determine the algorithm stop).
+     * The small i appearing at the end of the line stands for "infeasible" and will disappear only once Violated is 0.
+     *
+     * @param level verbosity level
+     */
+    void set_verbosity(unsigned int level)
+    {
+        m_verbosity = level;
+    };
+    /// Gets the verbosity level
+    unsigned int get_verbosity() const
+    {
+        return m_verbosity;
+    }
+    /// Gets the perturbation vector
+    const vector_double &get_perturb() const
+    {
+        return m_perturb;
+    }
+    /// Sets the perturbation vector
+    void set_perturb(const vector_double &perturb)
+    {
+        m_perturb = perturb;
+    }
+    /// Get log
+    /**
+     * A log containing relevant quantities monitoring the last call to evolve. Each element of the returned
+     * <tt> std::vector </tt> is a mbh::log_line_type containing: Fevals, Best, Violated, Viol.Norm and
+     * Trial as described in mbh::set_verbosity
+     * @return an <tt> std::vector </tt> of mbh::log_line_type containing the logged values Fevals,
+     * Violated, Viol.Norm and
+     * Trial
+     */
+    const log_type &get_log() const
+    {
+        return m_log;
+    }
+    /// Algorithm name
+    std::string get_name() const
+    {
+        return "Monotonic Basin Hopping (MBH) - Generalized";
+    }
+    /// Extra informations
+    std::string get_extra_info() const
+    {
+        std::ostringstream ss;
+        stream(ss, "\tStop: ", m_stop);
+        stream(ss, "\n\tPerturbation vector: ", m_perturb);
+        stream(ss, "\n\tSeed: ", m_seed);
+        stream(ss, "\n\tVerbosity: ", m_verbosity);
+        stream(ss, "\n\n\tInner algorithm: ", static_cast<const algorithm *>(this)->get_name());
+        stream(ss, "\n\tInner algorithm extra info: ");
+        stream(ss, "\n", static_cast<const algorithm *>(this)->get_extra_info());
+        return ss.str();
+    }
+    /// Serialization
+    template <typename Archive>
+    void serialize(Archive &ar)
+    {
+        ar(cereal::base_class<algorithm>(this), m_stop, m_perturb, m_e, m_seed, m_verbosity, m_log);
+    }
+
+private:
+    // Delete all that we do not want to inherit from problem
+    // A - Common to all meta
+    template <typename T>
+    bool is() const = delete;
+    bool has_set_seed() const = delete;
+    bool is_stochastic() const = delete;
+    bool has_set_verbosity() const = delete;
+
+#if __GNUC__ > 4
+    // NOTE: We delete the streaming operator overload called with mbh, otherwise the inner algo would stream
+    // NOTE: If a streaming operator is wanted for this class remove the line below and implement it.
+    friend std::ostream &operator<<(std::ostream &, const mbh &) = delete;
+#endif
+
+    unsigned int m_stop;
+    // The member m_perturb is mutable as to allow to construct mbh also using a perturbation defined as a scalar
```

Comment:
I also do not like this behaviour ... but could not come with something better. Maybe your suggestion would improve ... not sure.. will think about it. Leaving as it is for the moment